h2o-core/src/test/java/water/jdbc/SQLManagerTest.java

```diff
@@ -0,0 +1,127 @@
+package water.jdbc;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.ProvideSystemProperty;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import water.H2O;
+import water.H2ONode;
+import water.Paxos;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+public class SQLManagerTest {
+
+  @Rule
+  public final ProvideSystemProperty provideSystemProperty =
+      new ProvideSystemProperty(H2O.OptArgs.SYSTEM_PROP_PREFIX + "sql.connections.max", "7");
+
+
+  @Rule
+  public final RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+  private Field cloudSizeField;
+  private Field currentRuntimeField;
+
+  @Before
+  public void setUp() throws Exception {
+    Paxos._commonKnowledge = true;
+    cloudSizeField = H2O.class.getField("_memary");
```

Comment:
is all this reflection setup really necessary?
after all, SqlManager uses only H2O.getCloudSize() and Runtime.getRuntime().availableProcessors().
If you're able to mock the latter in the tests below, this should work for the former as well, no?

Reply:
You are right. There was PowerMockito used already partially. Direct reflection manipulation has been removed and replaced by PowerMockito. Of course, there are still assertions if given values are mocked exactly as planned.